File: Leetcode/Queue_Stack/232_Queue_using_stacks.cs
public class MyQueue
{
    //stack is LIFO, Queue is FIFO. Need to use to stacks, push twice,
    // so LIFO -> FOFI -> FIFO
    //https://leetcode.com/problems/implement-queue-using-stacks/solution/

    Stack stack = new Stack();
    Stack queue = new Stack();
    /** Push element x to the back of queue. */
    public void Push(int x)
    {
        stack.Push(x);
    }

    /** Removes the element from in front of queue and returns that element. */
    public int Pop()
    {
        Peek();
        return Convert.ToInt32(queue.Pop());
    }

    /** Get the front element. */
    public int Peek()
    {
        if (queue.Count == 0)
        {
            while (stack.Count != 0)
            {
                queue.Push(stack.Pop());
            }
        }
        return Convert.ToInt32(queue.Peek());

    }

    /** Returns whether the queue is empty. */
    public bool Empty()
    {
        return (stack.Count == 0 && queue.Count == 0);
    }
}

File: Leetcode/Queue_Stack/232_Queue_using_stacks.js

var MyQueue = function() {
    this.first = [];
    this.last = [];
};

MyQueue.prototype.push = function(x) {
   const length = this.first.length;
    for (let i = 0; i < length; i++) {
      this.last.push(this.first.pop());
    }
    this.last.push(x);
    return this;
};

MyQueue.prototype.pop = function() {
   const length = this.last.length;
    for (let i = 0; i < length; i++) {
      this.first.push(this.last.pop());
    }
  return this.first.pop() || null;
};

MyQueue.prototype.peek = function() {
    if (this.last.length > 0) {
      return this.last[0];
    }
    return this.first[this.first.length - 1] || null;
};

MyQueue.prototype.empty = function() {
  return (this.last.length == 0 && this.first.length == 0)
};
